src/main.rs

```diff
@@ -1,16 +1,41 @@
 mod api;
+pub mod batcher;
 mod models;
 use crate::api::api::run_api;
+use crate::batcher::batcher::batch_process;
 use crate::models::OrderBook;
-use tokio::select;
+use std::time::Duration;
+use tokio::time::delay_for;
+use tokio::{select, spawn};
 
 #[tokio::main]
 async fn main() {
     let orderbook = OrderBook::new();
-    let handler_api = run_api(orderbook);
+    let handler_api = run_api(orderbook.clone());
+    let handler_driver = run_driver(orderbook);
+
     select! {
-        err = handler_api => {
-            println!("run_api returned the following error {:?}", err);
+        e = handler_api => {
+            println!("run_api returned  {:?}", e);
+        },
+        e = handler_driver => {
+            println!("handler_driver returned {:?}", e);
         }
     }
 }
+
+async fn run_driver(orderbook: OrderBook) {
+    loop {
+        let orderbook_for_iteration = orderbook.clone();
+        spawn(async move {
```

Comment:
Why do you want to run this in a separate thread? Would you expect multiple run loops to be ongoing in parallel? Shouldn't one loop finish before we attempt another one?

Reply:
Would you expect multiple run loops to be ongoing in parallel?
Loops might not finish quickly, as we are waiting for transactions to be confirmed. The waiting for the mining will most likely block them quite a time. Hence, I think it makes sense to start them over, before they are officially ended.

I am using tokio::spawn and not thread::spawn, so it is really just another future.  But you are right that we might not even need tokio, I will try to rewrite this PR with it.

Reply:
You are right that tx confirmation might take some time. How do you envision the "starting over"? Would we try to settle the same orders as in the currently pending settlement (outbidding the old tx with the same nonce). 
Or would we optimistically assume the previous tx went through and resend a new settlement with a higher nonce?

Reply:
I think that settling the same orders does not make too much sense. Here, I would prefer a good "tx retry mechanism", as we have for the submit_solution already.

Yes, I would propose that the old/pending async task marks the orders used in their submission and gives the new async task the chance to recognize already used orders, and hence preventing double submission.